src/pudl/ferc_to_sqlite/cli.py

```diff
@@ -87,23 +87,42 @@ def parse_command_line(argv):
 
 
 def ferc_to_sqlite_job_factory(
-    logfile: str | None = None, loglevel: str = "INFO"
+    logfile: str | None = None,
+    loglevel: str = "INFO",
+    enable_xbrl: bool = True,
+    enable_dbf: bool = True,
 ) -> Callable[[], JobDefinition]:
     """Factory for parameterizing a reconstructable ferc_to_sqlite job.
 
     Args:
         loglevel: The log level for the job's execution.
         logfile: Path to a log file for the job's execution.
+        enable_xblr: if True, include XBRL data processing in the job.
+        enable_dbf: if True, include DBF data processing in the job.
 
     Returns:
         The job definition to be executed.
     """
+    if not (enable_xbrl or enable_dbf):
```

Comment:
The way we're splitting up or specifying which data to translate into SQLite feels messy to me in general. Not sure if there's a better way to do it though.

Right now in the ETL settings we ask the user for datasets and years, and then figure out whether that means translating from XBRL or DBF internally, which seems correct -- ideally the user wouldn't need to know which source the data is coming from.  But with 5 FERC forms (1, 2, 6, 60, 714) involved and 2 data formats for each of them (though the older 714 data isn't distributed as DBF for some reason) it feels like having some clear enumeration of the chunks of work that are going to get done internally would be more readable than having this high level job switch that says do DBF or do XBRL, and then another lower layer that's only visible in the settings that says which datasets to translate, and then yet another layer within the settings beneath that which indicates which partitions (years) to translate.

I guess what I'm imagining is that instead of just pulling the DBF vs. XBRL processing up to this high level, it might also make sense to pull both the dataset and data format up to this level and treat each dataset + data format as a separate job, since they'll each result in a separate SQLite output, and in a development context we'll often just want to re-run one of them in isolation. What do you think?

Reply:
Yeah. I agree that having one job per dataset/format combination would be the most natural structure. Let me refactor this accordingly.

Reply:
After some early tinkering with this, I feel like I will need to dig deeper into dagster and how it works and how to properly structure this. I think that it should be possible to auto generate some of these dagster ops/graphs programmatically which should reduce the boilerplate, but it might take a little extra effort and care to do so.

My suggestion would be to proceed with this PR as-is and follow up with cleanup/refactoring independently of this. What do you think about that? I can file an issue to track this.

Reply:
Sure, I think that's beyond the scope of this PR for right now, but it'd be good to make it more dynamic / granular at some point.  And we don't extract the FERC DBs very frequently, so this won't come up too often.